### PR TITLE
FIRECERT-2100: [POC] FCA implementation for user interest scenarios

### PR DESCRIPTION
**Pull request by Eswar2103**
# Description
Enter a detailed description of the PR here, including the nature and purpose of the changes.

## JIRA Ticket
Include a link to the JIRA Ticket linked to this PR.
[FIRECERT-2100](https://ccp.sys.comcast.net/browse/FIRECERT-2100)

## Additional Info (Test steps / Steps to reproduce etc.)
Details on how to test the feature

### CODE REVIEW

File: src/providers/UserInterestProvider.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { testDataHandler } from '../utils/Utils';
+
+export default class UserIntrestProvider {
+  userInterest() {
+    const ENTITYINFOEXAMPLES = testDataHandler('content', 'CONTENT');
+    if (ENTITYINFOEXAMPLES.hasOwnProperty(process.env.entityInfoKey)) {
```
Comment (Eswar2103) [inline src/providers/UserInterestProvider.js:23]: the response of **Content.requestUserInterest()** is expected to be in format of **EntityDetails**, there is already data in this **EntityDetails** format for federeated data, just using of same data, from FCS before operation will send the keyname of which to send back as response, that value will be send back from here

Comment (kevinshahfws) [reply src/providers/UserInterestProvider.js:23]: Is the work from FCS side is done?

File: src/App.js [hunk 2]
```diff
@@ -28,9 +28,11 @@ import { CONSTANTS } from './constant';
 require('dotenv').config({ override: true });
 import { checkMockOSRestInterface, TRUE_VALUES, getCurrentAppID } from './utils/Utils';
 import { AcknowledgeChallenge, Keyboard, PinChallenge } from '@firebolt-js/manage-sdk';
+import { Discovery } from '@firebolt-js/sdk';
 import PinChallengeProviderDelegater from './providers/PinChallengeDelegater';
 import KeyboardProviderDelegater from './providers/KeyboardProviderDelegater';
 import AckChallengeProviderDelegater from './providers/AckChallengeDelegater';
+import UserIntrestDelegater from './providers/UserInterestDelegater';
 const logger = require('./utils/Logger')('App.js');
```
Comment (neeradanelxsi) [inline src/App.js:36]: spelling is wrong for UserIntrestDelegater

Comment (Eswar2103) [reply src/App.js:35]: corrected

File: src/providers/UserInterestProvider.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      const USERINTERESTDATA = require('../../plugins/userInterest.json');
+      if (USERINTERESTDATA.hasOwnProperty(process.env.userInterestKey)) {
+        const data = USERINTERESTDATA[process.env.userInterestKey];
+        return Promise.resolve(data);
+      } else {
+        return Promise.resolve(null);
+      }
+    } catch (err) {
+      logger.error('Test data repo error: ', err);
+    }
```
Comment (neeradanelxsi) [inline src/providers/UserInterestProvider.js:31]: wouldn't it be better to keep the userInterest data in FCA itself and not parent?

File: src/App.js [hunk 4]
```diff
@@ -198,14 +199,16 @@ export default class App extends Base {
...
-            if (systemui) {
-              try {
+            try {
+              if (systemui) {
                 AcknowledgeChallenge.provide('xrn:firebolt:capability:usergrant:acknowledgechallenge', new AckChallengeProviderDelegater(this));
                 Keyboard.provide('xrn:firebolt:capability:input:keyboard', new KeyboardProviderDelegater(this));
                 PinChallenge.provide('xrn:firebolt:capability:usergrant:pinchallenge', new PinChallengeProviderDelegater(this));
-              } catch (err) {
-                logger.error('Could not set up providers' + err, 'LoadedState');
+              } else {
```
Comment (kevinshahfws) [inline src/App.js:207]: As discussed in the thread, if we can create a function that can decide to provide/not provide support for the provider based on the flag coming in Parameter.initialization or onNavigateTo via intent

File: src/providers/UserInterestProvider.js (hunk 3 above)
Comment (kevinshahfws) [reply src/providers/UserInterestProvider.js:31]: yes only if it doesn't contain any proprietary information

**Review by kevinshahfws — CHANGES_REQUESTED** [review]
Left few comments

File: src/providers/UserInterestProvider.js (hunk 1 above)
Comment (Eswar2103) [reply src/providers/UserInterestProvider.js]: yes

File: src/App.js (hunk 4 above)
Comment (Eswar2103) [reply src/App.js:207]: Is that ok we can handle this point and 
2.Doesn't register as a provider (platform should respond with not available)
I believe we don't have this support for any provider as of now but will have to design it
3. Registers as a provider but just never responds (platform should timeout the provider and respond with not available)
In new ticket https://ccp.sys.comcast.net/browse/FIRECERT-2148?

File: src/invokers/DiscoveryInvoker.js [hunk 5]
```diff
@@ -17,7 +17,7 @@
  */
 
 import { Discovery } from '@firebolt-js/sdk';
-// import { Content } from '@firebolt-js/discovery-sdk'
+// import { Content } from '@firebolt-js/discovery-sdk';
```
Comment (jbigel) [inline src/invokers/DiscoveryInvoker.js:20]: Can we remove this line?

File: src/providers/UserInterestProvider.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    try {
+      const USERINTERESTDATA = require('../source/userInterest.json');
+      if (USERINTERESTDATA.hasOwnProperty(process.env.userInterestKey)) {
+        const data = USERINTERESTDATA[process.env.userInterestKey];
+        return Promise.resolve(data);
+      } else {
+        return Promise.resolve(null);
+      }
+    } catch (err) {
+      logger.error('Test data repo error: ', err);
```
Comment (jbigel) [inline src/providers/UserInterestProvider.js:30]: Can this error message be informative on where the error occurred?

File: src/invokers/DiscoveryInvoker.js (hunk 5 above)
Comment (Eswar2103) [reply src/invokers/DiscoveryInvoker.js]: done

File: src/providers/UserInterestProvider.js (hunk 6 above)
Comment (Eswar2103) [reply src/providers/UserInterestProvider.js]: corrected

**Review by neeradanelxsi — APPROVED** [review]

**Review by kevinshahfws — APPROVED** [review]

**Pull request merged**